ZED Wrapper Build Error

### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues.
- [X] This issue is not a question, feature request, or anything other than a bug report directly related to this project.

### Description

When building the ROS2 wrapper in a Docker container, I get a build error. 

### Steps to Reproduce

Build using the following Dockerfile:
```
FROM stereolabs/zed:4.0-devel-cuda12.1-ubuntu20.04 as zed
SHELL ["/bin/bash", "-c"]

# installing essential dependencies
RUN apt-get update -y --allow-insecure-repositories && apt-get install curl -y
RUN curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
RUN apt-get update -y
RUN apt-get install -y \
    build-essential \
    cmake \
    git \
    python-is-python3 \
    python3-pip \
    nano \
    mesa-utils \
    ros-foxy-ros-base \
    python3-argcomplete \
    ros-dev-tools

# ros2 dependencies + others
RUN apt-get install --allow-unauthenticated -y \
    ros-foxy-xacro \
    ros-foxy-angles \
    ros-foxy-rviz-visual-tools \
    ros-foxy-joint-state-publisher \
    ros-foxy-joint-state-publisher-gui \
    ros-foxy-ament-cmake-clang-format \
    ros-foxy-rosidl-generator-cpp \
    ros-foxy-rosidl-default-generators \
    ros-foxy-image-transport \
    ros-foxy-ament-lint-auto \
    ros-foxy-ament-cmake-copyright \
    ros-foxy-ament-cmake-cppcheck \
    ros-foxy-ament-cmake-lint-cmake \
    ros-foxy-ament-cmake-pep257 \
    ros-foxy-ament-cmake-uncrustify \
    ros-foxy-ament-cmake-xmllint \
    ros-foxy-nmea-msgs \
    ros-foxy-graph-msgs \
    ros-foxy-diagnostic-updater \
    ros-foxy-geographic-msgs \
    ros-foxy-robot-localization \
    libopencv-dev \
    python3-opencv

# clone zed_ros2_wrapper and build
WORKDIR /root/dev_ws/src
RUN git clone --recursive https://github.com/stereolabs/zed-ros2-wrapper.git
WORKDIR /root/dev_ws
RUN rosdep init
RUN rosdep update
RUN rosdep install --from-paths src --ignore-src -r -y
RUN source /opt/ros/foxy/setup.bash && colcon build --symlink-install --event-handlers console_direct+ --base-paths src \
  --cmake-args ' -DCMAKE_BUILD_TYPE=Release' \
  ' -DCMAKE_LIBRARY_PATH=/usr/local/cuda/lib64/stubs' \
  ' -DCMAKE_CXX_FLAGS="-Wl,--allow-shlib-undefined"' --parallel-workers $(nproc)
RUN echo "source /opt/ros/foxy/setup.bash; source /root/dev_ws/install/setup.sh" >> ~/.bashrc
```

### Expected Result

The build should complete without error.

### Actual Result

The build fails. This is not the same error as #169. The build succeeds if checking out to commit `1b8e1ae9e42f24b9bf4843f2022c52125c622a3c`. Looking at the comparison, we see that these lines are the culprit for the error: [commit comparison](https://github.com/stereolabs/zed-ros2-wrapper/compare/1b8e1ae..c0e8d6b#diff-ec9dcfec438f3fa34a30946bd0cca4166788af7a65ff35e04681730555a0c931R465-R468).

Full log attached here: [build.log](https://github.com/stereolabs/zed-ros2-wrapper/files/12223886/build.log)

The main error in the log is as follows:
```
In file included from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/src/zed_camera_component.cpp:20:
/root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/zed_camera_component.hpp:466:40: error: use of deleted function ‘std::atomic<short unsigned int>::at
  466 |   std::atomic_uint16_t mAiInstanceID = 0;
      |                                        ^
In file included from /usr/include/c++/9/future:42,
                 from /opt/ros/foxy/include/tf2_ros/async_buffer_interface.h:34,
                 from /opt/ros/foxy/include/tf2_ros/buffer.h:39,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/sl_types.hpp:18,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/zed_camera_component.hpp:22,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/src/zed_camera_component.cpp:20:
/usr/include/c++/9/atomic:732:7: note: declared here
  732 |       atomic(const atomic&) = delete;
      |       ^~~~~~
/usr/include/c++/9/atomic:736:17: note:   after user-defined conversion: ‘constexpr std::atomic<short unsigned int>::atomic(std::atomic<short unsigned int>::__integral_type
  736 |       constexpr atomic(__integral_type __i) noexcept : __base_type(__i) { }
      |                 ^~~~~~
make[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:89: CMakeFiles/zed_camera_component.dir/src/zed_camera/src/zed_camera_component.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:79: CMakeFiles/zed_camera_component.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
--- stderr: zed_components
In file included from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/zed_camera_component.hpp:19,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/src/zed_camera_component.cpp:20:
/usr/local/zed/include/sl/Fusion.hpp:68:64: note: #pragma message: ~ FUSION SDK is distributed in Early Access ~
   68 | #pragma message("~ FUSION SDK is distributed in Early Access ~")
      |                                                                ^
In file included from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/src/zed_camera_component.cpp:20:
/root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/zed_camera_component.hpp:466:40: error: use of deleted function ‘std::atomic<short unsigned int>::at
  466 |   std::atomic_uint16_t mAiInstanceID = 0;
      |                                        ^
In file included from /usr/include/c++/9/future:42,
                 from /opt/ros/foxy/include/tf2_ros/async_buffer_interface.h:34,
                 from /opt/ros/foxy/include/tf2_ros/buffer.h:39,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/sl_types.hpp:18,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/include/zed_camera_component.hpp:22,
                 from /root/dev_ws/src/zed-ros2-wrapper/zed_components/src/zed_camera/src/zed_camera_component.cpp:20:
/usr/include/c++/9/atomic:732:7: note: declared here
  732 |       atomic(const atomic&) = delete;
      |       ^~~~~~
/usr/include/c++/9/atomic:736:17: note:   after user-defined conversion: ‘constexpr std::atomic<short unsigned int>::atomic(std::atomic<short unsigned int>::__integral_type
  736 |       constexpr atomic(__integral_type __i) noexcept : __base_type(__i) { }
      |                 ^~~~~~
make[2]: *** [CMakeFiles/zed_camera_component.dir/build.make:89: CMakeFiles/zed_camera_component.dir/src/zed_camera/src/zed_camera_component.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:79: CMakeFiles/zed_camera_component.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< zed_components [9.09s, exited with code 2]
```

### ZED Camera model

ZED2i

### Environment

```shell
OS: Ubuntu 22.04 (Ubuntu 20.04 in container)
CPU: AMD Ryzen Threadripper PRO 5995WX 64-Cores
GPU: 2x RTX 4090s
Zed SDK version: 4.0.5 (base image specified in the Dockerfile)
ROS Distro: ROS2 Foxy
```


### Anything else?
In the Dockerfile above, adding the below two lines to checkout this commit yields a successful build.
```
# previous lines...
WORKDIR /root/dev_ws/src
RUN git clone --recursive https://github.com/stereolabs/zed-ros2-wrapper.git

# ADD THESE LINES
##############
WORKDIR zed-ros2-wrapper
RUN git checkout 1b8e1ae9e42f24b9bf4843f2022c52125c622a3c
##############
```

--- a/zed_components/src/zed_camera/include/zed_camera_component.hpp
+++ b/zed_components/src/zed_camera/include/zed_camera_component.hpp
@@ -463,7 +463,7 @@ private:
   bool mGnss2BaseTransfValid = false;
   bool mMap2UtmTransfValid = false;
 
-  std::atomic_uint16_t mAiInstanceID = 0;
+  std::atomic_uint16_t mAiInstanceID;
   uint16_t mObjDetInstID;
   uint16_t mBodyTrkInstID;
   // <---- TF Transforms Flags

--- a/zed_components/src/zed_camera/src/zed_camera_component.cpp
+++ b/zed_components/src/zed_camera/src/zed_camera_component.cpp
@@ -63,7 +63,6 @@ namespace stereolabs
 
 ZedCamera::ZedCamera(const rclcpp::NodeOptions & options)
 : Node("zed_node", options),
-  mDiagUpdater(this),
   mVideoQos(1),
   mDepthQos(1),
   mSensQos(1),
@@ -72,7 +71,9 @@ ZedCamera::ZedCamera(const rclcpp::NodeOptions & options)
   mObjDetQos(1),
   mBodyTrkQos(1),
   mClickedPtQos(1),
-  mGnssFixQos(1)
+  mGnssFixQos(1),
+  mAiInstanceID(0),
+  mDiagUpdater(this)
 {
   RCLCPP_INFO(get_logger(), "********************************");
   RCLCPP_INFO(get_logger(), "      ZED Camera Component ");

